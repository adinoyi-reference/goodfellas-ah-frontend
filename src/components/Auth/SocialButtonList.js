import React from 'react';
import facebook from '../../assets/facebook-image.png';
import google from '../../assets/google-image.png';
import twitter from '../../assets/twitter-image.png';
import SocialButton from './SocialButton';


const SocialButtonList = () => (
  <div className="social-icons">
    <SocialButton
      image={facebook}
      type="facebook"
      className="facebook"
    />
    <SocialButton
      image={google}
      type="google"
      className="google"
    />
    <SocialButton
      image={twitter}
      type="twitter"
      className="twitter"
    />
  </div>
);


export default SocialButtonList;
