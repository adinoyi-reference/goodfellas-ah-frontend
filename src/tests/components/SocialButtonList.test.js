import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SocialButtonList from '../../components/Auth/SocialButtonList';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <SocialButtonList />
    </Root>,
  );
});

afterEach(() => wrapped.unmount());

describe('SocialButtonList UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('SocialButtonList API', () => {
  describe('social authentication buttons', () => {
    test('navigate the user to the correct route', () => {
      expect(wrapped.find('a').first().prop('href')).toEqual('http://localhost:3000/api/auth/facebook');
      expect(wrapped.find('a').at(1).prop('href')).toEqual('http://localhost:3000/api/auth/google');
      expect(wrapped.find('a').at(2).prop('href')).toEqual('http://localhost:3000/api/auth/twitter');
    });
  });
});
