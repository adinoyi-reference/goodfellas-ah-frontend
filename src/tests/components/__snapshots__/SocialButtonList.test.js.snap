// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtonList UI render features container should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <SocialButtonList>
      <div
        className="social-icons"
      >
        <SocialButton
          className="facebook"
          image="facebook-image.png"
          type="facebook"
        >
          <div
            className="social-link"
          >
            <a
              href="http://localhost:5000/api/auth/facebook"
            >
              <div
                className="facebook"
              >
                <img
                  alt=""
                  src="facebook-image.png"
                />
              </div>
            </a>
          </div>
        </SocialButton>
        <SocialButton
          className="google"
          image="google-image.png"
          type="google"
        >
          <div
            className="social-link"
          >
            <a
              href="http://localhost:5000/api/auth/google"
            >
              <div
                className="google"
              >
                <img
                  alt=""
                  src="google-image.png"
                />
              </div>
            </a>
          </div>
        </SocialButton>
        <SocialButton
          className="twitter"
          image="twitter-image.png"
          type="twitter"
        >
          <div
            className="social-link"
          >
            <a
              href="http://localhost:5000/api/auth/twitter"
            >
              <div
                className="twitter"
              >
                <img
                  alt=""
                  src="twitter-image.png"
                />
              </div>
            </a>
          </div>
        </SocialButton>
      </div>
    </SocialButtonList>
  </Provider>
</_default>
`;
